src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -165,24 +165,20 @@ def extract_return_type_from_udtf_type_hints(
     elif return_type_hint is None:
         return None
     else:
-        # Vectorized UDTF
-        if not installed_pandas:
-            raise RuntimeError(
-                "No pandas detected in local environment, please install pandas to use vectorized UDTF"
-            )
-        elif typing.get_origin(return_type_hint) == PandasDataFrame:
-            return PandasDataFrameType(
-                col_types=[
-                    python_type_to_snow_type(x)[0]
-                    for x in typing.get_args(return_type_hint)
-                ],
-                col_names=output_schema,
-            )
-        elif return_type_hint is pandas.DataFrame:
-            return PandasDataFrameType(
-                []
-            )  # placeholder, indicating the return type is pandas DataFrame
-        elif return_type_hint is NoneType:
+        if installed_pandas:  # Vectorized UDTF
```

Comment:
Can you help me understand why this fixes the issue? Do we happen to go to `elif` when `if not installed_pandas` is `True`?

Reply:
This block was not what broke the tests. I refactored this part a bit so that when no pandas is installed and the return type is invalid (not a collection type and not a pandas DataFrame), user would hit the invalid type invalid error instead of the "pandas not installed" error. The one liner changes below are what broke the tests before.